assets/app/scripts/directives/oscFileInput.js

```diff
@@ -32,6 +141,19 @@ angular.module('openshiftConsole')
             Logger.error("Could not read file", e);
           };
           reader.readAsText(file);
+        }
+
+        function removeDropZoneClasses(){
+          $('.drag-and-drop-zone').removeClass("show-drag-and-drop-zone highlight-drag-and-drop-zone");
+        }
+
+        function removeDropZoneElements() {
+          $(dropZoneSelectorName).remove();
+        }
+
+        scope.$on('$destroy', function(){
+          $(dropZoneSelectorName).off();
+          $(document).off('drop').off('dragenter').off('dragover').off('dragleave');
```

Comment:
Can you give these namespaces so we don't accidentally remove other drag/drop event handlers that might be registered? Maybe `"drop.oscFileInput"` when you call `on()`. Then `$(document).off('.oscFileInput')` should be enough to unregister all the different event types (untesed).

http://api.jquery.com/on/
http://api.jquery.com/off/